honeybee_energy/properties/face.py

```diff
@@ -0,0 +1,116 @@
+
+    def __init__(self, host, construction=None):
+        """Initialize Face energy properties.
+
+        Args:
+            host: A honeybee_core Face object that hosts these properties.
+            construction: An optional Honeybee OpaqueConstruction object for
+                the face. If None, it will be set by the parent Room ConstructionSet
+                or the the Honeybee default generic ConstructionSet.
+        """
+        self._host = host
+        self.construction = construction
+
+    @property
+    def host(self):
+        """Get the Face object hosting these properties."""
+        return self._host
+
+    @property
+    def construction(self):
+        """Get or set Face Construction.
+
+        If the Construction is not set on the face-level, then it will be assigned
+        based on the ConstructionSet assigned to the parent Room.  If there is no
+        parent Room or the the parent Room's ConstructionSet has no construction for
+        the Face type and boundary_condition, it will be assigned using the honeybee
+        default generic construction set.
+        """
+        if self._construction:  # set by user
+            return self._construction
+        elif self._host.has_parent:  # set by parent zone
+            return self._host.parent.construction_set.get_face_construction(
+                self._host.type.name, self._host.boundary_condition.name)
+        else:
+            return generic_costruction_set.get_face_construction(
+                self._host.type.name, self._host.boundary_condition.name)
+
+    @construction.setter
+    def construction(self, value):
+        if value is not None:
+            assert isinstance(value, OpaqueConstruction), \
+                'Expected Opaque Construction for face. Got {}'.format(type(value))
+            value.lock()  # lock editing in case construction has multiple references
+        self._construction = value
+
+    @property
+    def is_construction_set_by_user(self):
+        """Check if construction is set by user."""
+        return self._construction is not None
+
+    @property
+    def materials(self):
+        """List of materials in the face construction."""
+        return self.construction.materials
+
+    @property
+    def r_value(self):
+        """R-value of the Face [m2-K/W] (excluding air films)."""
+        return self.construction.r_value
+
+    @property
+    def u_value(self):
+        """U-value of the Face [W/m2-K] (excluding air films)."""
+        return self.construction.u_value
+
+    @property
+    def r_factor(self):
+        """Face R-factor [m2-K/W] (including standard resistances for air films)."""
+        return self.construction.r_factor
+
+    @property
+    def u_factor(self):
+        """Face U-factor [W/m2-K] (including standard resistances for air films)."""
+        return self.construction.u_factor
+
+    def to_dict(self):
+        """Return energy properties as a dictionary."""
+        base = {'energy': {}}
```

Comment:
Shouldn't it have the {'type': 'FaceEnergyProperties'} at least?

Reply:
Sounds good to me.  I just kept it like this because this is how you set it up originally.  I'll add the type for all of the Energy Property objects.